editors/sc-ide/widgets/settings/sclang_page.cpp

```diff
@@ -196,8 +199,13 @@ void SclangPage::readLanguageConfig() {
 }
 
 void SclangPage::writeLanguageConfig() {
-    if (!sclangConfigDirty)
+    if (!sclangConfigDirty) {
+        if (sclangConfigChanged) {
+            dialogConfigFileUpdated();
+            sclangConfigChanged = false; // avoid double invocation
```

Comment:
how could this trigger a double invocation? Can `writeLanguageConfig` be invoked multiple times?

Reply:
I don't know. But I observed double invocation. And also the flag should be reset anyway, no?

Reply:
Oh right - we have an *Apply* button in which case it should be indeed retested as the widget would not be destroyed in this case.